plugins/modules/netbox_fhrp_group.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, Andrii Konts (@andrii-konts) <andrew.konts@uk2group.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: netbox_fhrp_group
+short_description: Create, update or delete FHRP groups within NetBox
+description:
+  - Creates, updates or removes FHRP groups from NetBox
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Andrii Konts (@andrii-konts)
+requirements:
+  - pynetbox
+version_added: '3.12.0'
+extends_documentation_fragment:
+  - netbox.netbox.common
+options:
+  data:
+    type: dict
+    description:
+      - Defines the FHRP group configuration
+    suboptions:
+      protocol:
+        description:
+          - Protocol
+        required: False
+        type: str
+        choices:
+          - vrrp2
+          - vrrp3
+          - carp
+          - clusterxl
+          - hsrp
+          - glbp
+          - other
+      group_id:
+        description:
+          - Group ID (0 .. 32767)
+        type: int
+        required: true
+      auth_type:
+        description:
+          - Authentication type
+        choices:
+          - plaintext
+          - md5
+        type: str
+      auth_key:
+        description:
+          - Authentication key (max length 255)
+        type: str
+      description:
+        description:
+          - Description (max length 200)
+        required: false
+        type: str
+      tags:
+        description:
+          - Any tags that the FHRP group may need to be associated with
+        required: false
+        type: list
+        elements: raw
+      custom_fields:
+        description:
+          - Must exist in NetBox
+        required: false
+        type: dict
+    required: true
+"""
+
+EXAMPLES = r"""
+- name: "Test NetBox modules"
+  connection: local
+  hosts: localhost
+  gather_facts: False
+
+  tasks:
+    - name: "Create FHRP group within netbox"
+      netbox_fhrp_group:
+        netbox_url: http://netbox.local
+        netbox_token: thisIsMyToken
```

Comment:
Could you also modify these usage examples to use the default module group?

https://netbox-ansible-collection.readthedocs.io/en/latest/getting_started/how-to-use/advanced.html#using-module-defaults-groups

Reply:
Usage examples have been modified, but module_defaults won't work for ansible-core < 2.12

Reply:
All ansible-core releases =< 2.12 is end of life so that's fine.